src/Configuration/Config/src/ConfigurationKeyComparer.cs

```diff
@@ -26,8 +26,8 @@ public class ConfigurationKeyComparer : IComparer<string>
         /// <returns></returns>
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
-            var yParts = y?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
+            var xParts = x?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
```

Comment:
Is it a common case that either `x` or `y` is `null`?

If so, it may be better to hold `Array.Empty<string>` in a field, as the call to `Empty` won't be inlined (last time I checked this was true).

If not, nevermind this comment.